yarn-project/end-to-end/src/e2e_synching.test.ts

```diff
@@ -406,89 +429,256 @@ describe('e2e_synching', () => {
+        async (opts: Partial<EndToEndContext>, variant: TestVariant) => {
+          const rollup = getContract({
+            address: opts.deployL1ContractsValues!.l1ContractAddresses.rollupAddress.toString(),
+            abi: RollupAbi,
+            client: opts.deployL1ContractsValues!.walletClient,
+          });
 
-  it('replay, then prune and only then perform an initial sync', async () => {
-    if (AZTEC_GENERATE_TEST_DATA) {
-      return;
-    }
+          const pendingBlockNumber = await rollup.read.getPendingBlockNumber();
+          await rollup.write.setAssumeProvenThroughBlockNumber([pendingBlockNumber - BigInt(variant.blockCount) / 2n]);
 
-    const variant = variants[0];
+          const timeliness = (await rollup.read.EPOCH_DURATION()) * 2n;
+          const [, , slot] = await rollup.read.blocks([(await rollup.read.getProvenBlockNumber()) + 1n]);
+          const timeJumpTo = await rollup.read.getTimestampForSlot([slot + timeliness]);
 
-    const beforeSync = async (opts: Partial<EndToEndContext>) => {
-      const rollup = getContract({
-        address: opts.deployL1ContractsValues!.l1ContractAddresses.rollupAddress.toString(),
-        abi: RollupAbi,
-        client: opts.deployL1ContractsValues!.walletClient,
-      });
+          await opts.cheatCodes!.eth.warp(Number(timeJumpTo));
+
+          const watcher = new AnvilTestWatcher(
+            opts.cheatCodes!.eth,
+            opts.deployL1ContractsValues!.l1ContractAddresses.rollupAddress,
+            opts.deployL1ContractsValues!.publicClient,
+          );
+          await watcher.start();
+
+          const aztecNode = await AztecNodeService.createAndSync(opts.config!, new NoopTelemetryClient());
+          const sequencer = aztecNode.getSequencer();
+
+          const blockBeforePrune = await aztecNode.getBlockNumber();
+
+          await rollup.write.prune();
 
-      const pendingBlockNumber = await rollup.read.getPendingBlockNumber();
-      await rollup.write.setAssumeProvenThroughBlockNumber([pendingBlockNumber - BigInt(variant.blockCount) / 2n]);
+          await sleep(5000);
+          expect(await aztecNode.getBlockNumber()).toBeLessThan(blockBeforePrune);
 
-      const timeliness = (await rollup.read.EPOCH_DURATION()) * 2n;
-      const [, , slot] = await rollup.read.blocks([(await rollup.read.getProvenBlockNumber()) + 1n]);
-      const timeJumpTo = await rollup.read.getTimestampForSlot([slot + timeliness]);
+          const { pxe } = await setupPXEService(aztecNode!);
+          variant.setPXE(pxe);
 
-      await opts.cheatCodes!.eth.warp(Number(timeJumpTo));
+          const blockBefore = await aztecNode.getBlock(await aztecNode.getBlockNumber());
 
-      await rollup.write.prune();
-    };
+          sequencer?.updateSequencerConfig({ minTxsPerBlock: variant.txCount, maxTxsPerBlock: variant.txCount });
+          const txs = await variant.createAndSendTxs();
```

Comment:
I'm kind of surprised this works.

The node has had its archive pruned, but the world state is still unpruned. So we would expect this to fail to prove/verify at the moment right? And the roots it is publishing are garbage?

Reply:
Look at the start of that test. 
```
 it.skip('node following prunes and can extend chain', async () => {
      // @todo This test is to be activated when we can unwind the world state
      // It will currently stall forever as the state will never match.
```